prow/github/fakegithub/fakegithub.go

```diff
@@ -507,6 +509,16 @@ func (f *FakeClient) CreateProjectCard(columnID int, projectCard github.ProjectC
 	}
 
 	for project, columnIDMap := range f.ColumnIDMap {
+		for id := range columnIDMap {
+			// Make sure that we behave same as github API
+			// Create project will generate an error when the card already exist in the project
+			if _, exists := columnIDMap[columnID]; exists {
+				card, err := f.GetColumnProjectCard(id, projectCard.ContentURL)
+				if err == nil && card != nil && *card == projectCard {
```

Comment:
I think there is still a collision even if `*card != projectCard`.

Reply:
There can be only one project card per issue/PR and since we are looking through all columns in a project to see if there exists a projectCard  we should be safe once this check passes. If you are worried about collision due to concurrency, we will have a failure from the CreateCard api itself.